stream/each.js

```diff
@@ -1,78 +1,24 @@
-
-var through = require('through2');
+const from = require('from2');
 
 /**
   query for each row in a table; one by one (with stream backpressure).
 **/
-function streamFactory( db, table, condition ){
-
-  // create prepared statement
-  var stmt = db.prepare( 'SELECT * FROM ' + table + ' WHERE id = ?' );
-  var readOne;
-
-  // create a passthrough stream which also requests a new record from the
-  // database after it passed the previous one downstream.
-  var stream = through.obj( function( row, _, next ){
-    this.push( row );
-    readOne();
-    next();
-  }, function flush( next ){
-    stmt.finalize(); // finalize prepared statement
-    next();
-  });
-
-  db.serialize( function(){
-
-    var cur = 1; // current row number
-    var max = 0; // maximum rowid in table
-
-    // read the next row from the table
-    readOne = function(){
-
-      // reached last record in table
-      if( cur > max ){ return stream.end(); }
-
-      // get record by id
-      stmt.get([ cur++ ], function( err, row ){
-
-        // an error occurred
-        if( err ){ console.error( err ); }
-
-        // id not found in table (possibly a deleted record?); continue to the next id
-        if( !row ){ return readOne(); }
-
-        // write row on to stream
-        stream.write( row );
-      });
-    };
-
-    // calculate the highest rowid in the table
-
-    var sql = 'SELECT MAX( rowid ) as max, MIN( rowid ) as min FROM ' + table;
-    if( 'string' === typeof condition && condition.length ){
-      sql += ' ' + condition;
-    }
-
-    db.get( sql, function( err, row ){
-
-      // an error occurred
-      if( err ){ return console.error( err ); }
-      if( !row || !row.max || !row.min ){
-        stream.end();
-        return console.error( 'no rows in table', table );
+function streamFactory(db, sql) {
+  const stmt = db.prepare(sql);
+  const iterator = stmt.iterate();
+
+  return from.obj((size, next) => {
+    var ok = true;
+    while (ok) {
+      const elt = iterator.next();
```

Comment:
elt?

Reply:
heya, I appreciate that your intention is to make the code more easily readable for new developers, however I don't think scrutinising the naming of every variable is a productive use of our time?

this variable was named in a community contribution via https://github.com/pelias/whosonfirst/pull/417.

we currently don't have a naming policy for javascript variables, the problem with these things is that everyone has their own opinion on what is a good or bad name (usually based on the programming languages they grew up with).

so, in lieu of a standard to adhere to, our general policy is to make PR reviews as easy as possible on the contributor by not sweating the minor details.

a quick google shows that this is a common variable name for iterators in the Java and Pascal languages.